packages/amplify-cli-core/src/errors/amplify-exception.ts

```diff
@@ -25,29 +25,20 @@ export abstract class AmplifyException extends Error {
   constructor(
     public readonly name: AmplifyExceptionType,
     public readonly classification: AmplifyExceptionClassification,
-    private readonly options: AmplifyExceptionOptions,
+    public readonly options: AmplifyExceptionOptions,
     public readonly downstreamException?: Error,
   ) {
     // If an AmplifyException was already thrown, we must allow it to reach the user.
     // This ensures that resolution steps, and the original error are bubbled up.
-    super(downstreamException instanceof AmplifyException ? downstreamException.message : options.message);
+    super(options.message);
 
     // https://github.com/Microsoft/TypeScript-wiki/blob/main/Breaking-Changes.md#extending-built-ins-like-error-array-and-map-may-no-longer-work
     Object.setPrototypeOf(this, AmplifyException.prototype);
 
-    if (downstreamException instanceof AmplifyException) {
-      this.stack = downstreamException.stack ? downstreamException.stack : this.stack;
-      this.message = downstreamException.message;
-      this.details = downstreamException.details;
-      this.resolution = downstreamException.resolution;
-      this.link = downstreamException.link;
-    } else {
-      this.stack = options.stack ? options.stack : this.stack;
-      this.message = options.message;
-      this.details = options.details;
-      this.resolution = 'resolution' in options ? options.resolution : undefined;
-      this.link = 'link' in options ? options.link : undefined;
-    }
+    this.message = options.message;
+    this.details = options.details;
+    this.resolution = 'resolution' in options ? options.resolution : undefined;
+    this.link = 'link' in options ? options.link : undefined;
```

Comment:
would it make sense to fall back to default link like in that factory method we have around ?

Reply:
we can discuss this offline, the idea is to enforce either passing link or resolution when this object is created, adding default defeats the purpose, but again we have `amplifyFaultWithTroubleshootingLink` and `amplifyErrorWithTroubleshootingLink` that already defeat that purpose, so maybe we should just have `link` and `resolution` optional and drop the intersection type and those 2 methods

Reply:
I think we should get rid of `WithTroubleshootingLink` and have ctor to capture all cases.
I can also imagine to pass both resolution and link. For example when I want to pass most common resolution and direct to link for other edge cases.

Reply:
Maybe it makes sense for resolution and link to both be optional in the input args but if link is not set it defaults to the troubleshooting link rather than undefined? That way link is guaranteed to always be set (although it may not be helpful for the given error) and the thrower can optionally override the link or set a resolution

Reply:
```
if ( ! resolution && ! link) {
 link = defaultLink;
}
```